ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/422__model_serving_llm.robot

```diff
@@ -866,33 +886,41 @@ Model Response Should Match The Expectation
-    [Arguments]    ${models_names}    ${namespace}    ${endpoint}=${CAIKIT_ALLTOKENS_ENDPOINT}    ${n_times}=10
-    ...            ${streamed_response}=${FALSE}    ${query_idx}=0
+    [Arguments]    ${model_name}    ${namespace}    ${isvc_name}=${model_name}    ${endpoint}=${CAIKIT_ALLTOKENS_ENDPOINT}    ${n_times}=10
+    ...            ${streamed_response}=${FALSE}    ${query_idx}=0    ${validate_response}=${TRUE}    &{args}
+    IF    ${validate_response} == ${FALSE}
+        ${skip_json_load_response}=    Set Variable    ${TRUE}
+    ELSE
+        ${skip_json_load_response}=    Set Variable    ${streamed_response}    # always skip if using streaming endpoint
+    END
+    ${host}=    Get KServe Inference Host Via CLI    isvc_name=${isvc_name}   namespace=${namespace}
+    ${body}=    Set Variable    '{"text": "${EXP_RESPONSES}[queries][${query_idx}][query_text]"}'
+    ${header}=    Set Variable    'mm-model-id: ${model_name}'
     FOR    ${counter}    IN RANGE    0    ${n_times}    1
         Log    ${counter}
-        FOR    ${index}    ${model_name}    IN ENUMERATE    @{models_names}
-            Log    ${index}: ${model_name}
-            ${host}=    Get KServe Inference Host Via CLI    isvc_name=${model_name}   namespace=${namespace}
-            ${body}=    Set Variable    '{"text": "${EXP_RESPONSES}[queries][${query_idx}][query_text]"}'
-            ${header}=    Set Variable    'mm-model-id: ${model_name}'
-            ${res}=    Query Model With GRPCURL   host=${host}    port=443
-            ...    endpoint=${endpoint}
-            ...    json_body=${body}    json_header=${header}
-            ...    insecure=${TRUE}    skip_res_json=${streamed_response}
-            Log    ${res}
+        ${res}=    Query Model With GRPCURL   host=${host}    port=443
+        ...    endpoint=${endpoint}
+        ...    json_body=${body}    json_header=${header}
+        ...    insecure=${TRUE}    skip_res_json=${skip_json_load_response}
+        ...    &{args}
+        Log    ${res}
+        IF    ${validate_response} == ${TRUE}
             Run Keyword And Continue On Failure
             ...    Model Response Should Match The Expectation    model_response=${res}    model_name=${model_name}
             ...    streamed_response=${streamed_response}    query_idx=${query_idx}
         END
     END
 
-Compile And Query LLM model
-    [Arguments]    ${isvc_name}     ${model_storage_uri}    ${model_name}
+Compile Deploy And Query LLM model
+    [Documentation]    Group together the test steps for preparing, deploying
+    ...                and querying a model
+    [Arguments]    ${model_storage_uri}    ${model_name}    ${isvc_name}=${model_name}
     ...            ${canaryTrafficPercent}=${EMPTY}   ${namespace}=${TEST_NS}  ${sa_name}=${DEFAULT_BUCKET_SA_NAME}
-    ...            ${multiple_query}=${EMPTY}
+    ...            ${n_queries}=${1}    ${query_idx}=${0}    ${validate_response}=${TRUE}
+    ${models_names}=    Create List    ${model_name}
```

Comment:
seems that this robocop alert is relevant? Variable assigned but not used -> should it be removed?

Reply:
yep, removed